pkg/vmprovider/providers/vsphere/contentlibrary/content_library_cache.go

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) 2023 VMware, Inc. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package contentlibrary
+
+import (
+	"sync"
+	"time"
+
+	"github.com/vmware/govmomi/ovf"
+)
+
+type Cache struct {
+	mu    sync.Mutex
+	items map[string]*cacheItem
+}
+
+type cacheItem struct {
+	ovfEnvelope    ovf.Envelope
```

Comment:
Consider setting a max number of items that can be cached, potentially the cache might take some good memory space. If there are 1000 items with 100kb OVF descriptor in the cluster, then the cache may take 100mb if they may be cached around the same time period.

Reply:
That's a fair point. I applied a max size and set it to 100 in the new patch.

Reply:
CL validates OVF descriptor to ensure less than 4mb per OVF spec, most OVF template size should be way less than that. Tracking the total size of all cache entries can be also helpful, so it can have the cache bounded by some fixed or configurable memory space as well.

Reply:
In the worst case then, the cache can grow up to 4MB * 100 = 400MB which is around 381MiB, less than the current 500MiB limit on the container. I checked the TKG CL and it shows as 4.23 MB in total. In case we encounter any OOM from this cache, we can configure a smaller max size and shorter expiration time to clean up items more frequently.

Reply:
I wouldn't expect that to leave sufficient for the pod to not OOM.

It does seem that eventually that a combination of large OVF image(s) and/or mapped to many namespaces, we'll be forced to do MD-only watches and disable the ctrl-runtime client cache for NS-scoped image. That said, at present, if this cache doesn't contain cluster-scoped images, it is more or less another namespace worth of images.